lib/open_project_scimitar/limited_service_provider_configuration.rb

```diff
@@ -28,32 +28,29 @@
 # See COPYRIGHT and LICENSE files for more details.
 #++
 
-module ScimV2
-  module ScimControllerMixins
-    def self.included(base)
-      base.prepend(Overwrites)
-    end
-
-    module Overwrites
-      # Completely overwriting authenticate method of Scimitar
-      def authenticate
-        if !EnterpriseToken.allows_to?(:scim_api) || !OpenProject::FeatureDecisions.scim_api_active?
-          return handle_scim_error(Scimitar::AuthenticationError.new)
-        end
-
-        User.current = warden.authenticate!(scope: :scim_v2)
+module OpenProjectScimitar
+  class LimitedServiceProviderConfiguration
+    include ActiveModel::Model
 
-        # Only a ServiceAccount associated with a ScimClient can use SCIM Server API
-        unless User.current.respond_to?(:service) && User.current.service.is_a?(ScimClient)
-          handle_scim_error(Scimitar::AuthenticationError.new)
-        end
-      end
+    attr_accessor(
+      :authenticationSchemes,
```

Comment:
I think rubocop complains about this not being called `authentication_schemes`. I am wondering: Is it required that the attribute accessor is camel cased?

Reply:
Eventually it should be camel cased in the returned json body.

Reply:
We do not. Sure, let's change it.